polaris-react/src/components/Filters/Filters.tsx

```diff
@@ -1,605 +1,338 @@
-      helpText,
-      hideTags,
-      hideQueryField,
-      disableQueryField = false,
-      i18n,
-      mediaQuery: {isNavigationCollapsed},
-    } = this.props;
-    const {resourceName} = this.context;
-    const {open, readyForFocus} = this.state;
+  const shouldShowAddButton = enabledFilters.some((filter) => !filter.pinned);
 
-    const backdropMarkup = open ? (
+  const additionalContent = useMemo(() => {
+    return (
       <>
-        <ScrollLock />
-        <div className={styles.Backdrop} onClick={this.closeFilters} />
-      </>
-    ) : null;
-
-    const filtersContentMarkup = filters.map((filter, index) => {
-      const filterIsOpen = this.state[`${filter.key}${Suffix.Filter}`] === true;
-      const icon = filterIsOpen ? ChevronUpMinor : ChevronDownMinor;
-      const className = classNames(
-        styles.FilterTriggerContainer,
-        filterIsOpen && styles.open,
-        index === 0 && styles.first,
-        filters.length !== 1 && index === filters.length - 1 && styles.last,
-      );
-
-      const appliedFilterContent = this.getAppliedFilterContent(filter.key);
-      const appliedFilterBadgeMarkup = appliedFilterContent ? (
-        <div className={styles.AppliedFilterBadgeContainer}>
-          <Badge status="new">{appliedFilterContent}</Badge>
+        <div className={styles.Spinner}>
+          {loading ? <Spinner size="small" /> : null}
         </div>
-      ) : null;
-
-      const collapsibleID = `${filter.key}Collapsible`;
-
-      const buttonClassName = classNames(styles.FilterTrigger);
-
-      return (
-        <div key={filter.key} className={className}>
-          <button
-            onClick={() => this.toggleFilter(filter.key)}
-            className={buttonClassName}
-            id={`${filter.key}ToggleButton`}
-            type="button"
-            aria-controls={collapsibleID}
-            aria-expanded={filterIsOpen}
-          >
-            <div className={styles.FilterTriggerLabelContainer}>
-              <h3 className={styles.FilterTriggerTitle}>
-                <Text
-                  as="span"
-                  color={
-                    this.props.disabled || filter.disabled
-                      ? 'subdued'
-                      : undefined
-                  }
-                >
-                  {filter.label}
-                </Text>
-              </h3>
-              <span className={styles.FilterTriggerIcon}>
-                <Icon source={icon} color="base" />
-              </span>
-            </div>
-            {appliedFilterBadgeMarkup}
-          </button>
-          <Collapsible
-            id={collapsibleID}
-            open={filterIsOpen}
-            onAnimationEnd={this.setReadyForFocus(true)}
+        {children}
+      </>
+    );
+  }, [loading, children]);
+
+  return (
+    <div
+      className={classNames(
+        styles.Filters,
+        hideQueryField && styles.hideQueryField,
+      )}
+    >
+      {hideQueryField ? null : (
```

Comment:
```suggestion
    const queryFieldMarkup = !hideQueryField
      ? {markup}
      : null;
      
     ...
      
     return (
       ...
         {queryFieldMarkup}
```

Let's move conditionals and other large blocks of markup out of the return value and into variables so it's easier to parse the different parts in the layout.